crates/uv-distribution/src/index/cached_wheel.rs

```diff
@@ -135,26 +131,21 @@ impl CachedWheel {
         }
 
         let Archive { id, hashes, .. } = archive;
-
         let entry = cache.entry(CacheBucket::Archive, "", id);
 
         // Convert to a cached wheel.
         Some(Self {
-            filename,
+            filename: archive.filename,
             entry,
             hashes,
             cache_info,
         })
     }
 
-    /// Read a cached wheel from a `.rev` pointer (e.g., `anyio-4.0.0-py3-none-any.rev`).
+    /// Read a cached wheel from a `.rev` pointer
     pub fn from_local_pointer(path: impl AsRef<Path>, cache: &Cache) -> Option<Self> {
         let path = path.as_ref();
 
-        // Determine the wheel filename.
-        let filename = path.file_stem()?.to_str()?;
-        let filename = WheelFilename::from_stem(filename).ok()?;
```

Comment:
@nkitsaini -- I also removed the `Version` checks from this method. I think the version (before the `-`) isn't actually guaranteed to be valid, since it could be truncated to, like... `1.2.0+`. I might change that in the cache key to avoid those ambiguities, but still seems safer to remove the assumption here. (We already validate that the file ends in `.rev` anyway.)

Reply:
Ah, nice. Missed that. I am also not sure how valuable these checks were. It would make sense to use cache even if the filename is wrong but it points to a valid archive. And I wouldn't assume there would be any invalid files unless created manually, which should be pretty rare hence negating any performance benefits from saved disk I/O.

Maybe I am missing something?